### PR TITLE
Update Webhook api calls

### LINKED ISSUE #183: Add support to filter hook list by meetingID
According to the API docs you can filter the list of the registered webhooks with a meetingID
See: https://docs.bigbluebutton.org/development/webhooks/#hookslist

Add feature in this API to use this filtering.

Should be done in 5.x as this is a BC.

### PR DESCRIPTION
**Pull request by SamuelWei**
Fixes #183

- Add new `HooksListParameters` class as required parameter for `hooksList()` and `getHooksListUrl()` (**Breaking Change**)  (see https://docs.bigbluebutton.org/development/webhooks/#hookslist)
- Add `getEventID()` / `setEventID(string $id)` on HooksCreateParameters  (see https://docs.bigbluebutton.org/development/webhooks/#hookscreate)
- Add `isMissingHook()` and `isHookError()` to HooksDestroyResponse (see https://docs.bigbluebutton.org/development/webhooks/#hooksdestroy)

### CODE REVIEW

**Comment by SamuelWei** [comment]
Should be part of 5.x due to the BC

**Review by FelixJacobi — APPROVED** [review]

**Comment by FelixJacobi** [comment]
Ok, we must start a 5.4 branch based 5.3 then and merge this up to master later after merging this PR to that one.

File: src/BigBlueButton.php [hunk 1]
```diff
@@ -418,14 +419,14 @@ public function hooksCreate(HooksCreateParameters $hookCreateParams): HooksCreat
...
 
-    public function getHooksListUrl(): string
+    public function getHooksListUrl(HooksListParameters $hooksListParameters): string
     {
-        return $this->urlBuilder->buildUrl(ApiMethod::HOOKS_LIST);
+        return $this->urlBuilder->buildUrl(ApiMethod::HOOKS_LIST, $hooksListParameters->getHTTPQuery());
     }
 
-    public function hooksList(): HooksListResponse
+    public function hooksList(HooksListParameters $hooksListParameters): HooksListResponse
```
Comment (FelixJacobi) [inline src/BigBlueButton.php:427]: Hm, that is a breaking change, this parameter must not be a real parameter in 5.x and this method must trigger a deprecation notice when it is missing.

File: src/BigBlueButton.php [hunk 2]
```diff
@@ -418,14 +419,14 @@ public function hooksCreate(HooksCreateParameters $hookCreateParams): HooksCreat
         return new HooksCreateResponse($xml);
     }
 
-    public function getHooksListUrl(): string
+    public function getHooksListUrl(HooksListParameters $hooksListParameters): string
```
Comment (FelixJacobi) [inline src/BigBlueButton.php:422]: Hm, that is a breaking change, this parameter must not be a real parameter in 5.x and this method must trigger a deprecation notice when it is missing.

File: tests/unit/Parameters/HooksListParametersTest.php [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+ *
+ * BigBlueButton is distributed in the hope that it will be useful, but WITHOUT ANY
+ * WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+ * PARTICULAR PURPOSE. See the GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License along
+ * with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace unit\Parameters;
```
Comment (FelixJacobi) [inline tests/unit/Parameters/HooksListParametersTest.php:20]: I guess, your IDE wasn't smart enough here ;) .

File: tests/unit/Parameters/HooksListParametersTest.php [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+ * with BigBlueButton; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace BigBlueButton\Parameters;
+
+use BigBlueButton\TestCase;
+
+final class HooksListParametersTest extends TestCase
+{
+    public function testHooksListParameters()
```
Comment (FelixJacobi) [inline tests/unit/Parameters/HooksListParametersTest.php:26]:
```suggestion
    public function testHooksListParameters(): void
```

File: src/Parameters/HooksListParameters.php [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+
+namespace BigBlueButton\Parameters;
+
+/**
+ * Class HooksListParameters.
+ *
+ * @method string getMeetingID()
+ * @method $this  setMeetingID(string $id)
+ */
+class HooksListParameters extends MetaParameters
```
Comment (FelixJacobi) [inline src/Parameters/HooksListParameters.php:28]:
```suggestion
final class HooksListParameters extends MetaParameters
```

File: tests/unit/Responses/HooksDestroyResponseTest.php [hunk 6]
```diff
@@ -49,4 +49,22 @@ public function testHooksDestroyResponseTypes()
...
+    public function testHookDestroyMissingHook()
+    {
+        $xml = simplexml_load_string('<response><returncode>FAILED</returncode><messageKey>destroyMissingHook</messageKey><message>The hook informed was not found.</message></response>');
+
+        $destroyResponse = new HooksDestroyResponse($xml);
+        $this->assertTrue($destroyResponse->failed());
+        $this->assertTrue($destroyResponse->isMissingHook());
+    }
+
+    public function testHookDestroyHookError()
```
Comment (FelixJacobi) [inline tests/unit/Responses/HooksDestroyResponseTest.php:62]:
```suggestion
    public function testHookDestroyHookError(): void
```

File: tests/unit/Responses/HooksDestroyResponseTest.php [hunk 7]
```diff
@@ -49,4 +49,22 @@ public function testHooksDestroyResponseTypes()
         $this->assertEachGetterValueIsString($this->destroyResponse, ['getReturnCode']);
         $this->assertEachGetterValueIsBoolean($this->destroyResponse, ['removed']);
     }
+
+    public function testHookDestroyMissingHook()
```
Comment (FelixJacobi) [inline tests/unit/Responses/HooksDestroyResponseTest.php:53]:
```suggestion
    public function testHookDestroyMissingHook(): void
```

**Review by FelixJacobi — APPROVED** [review]

**Pull request merged**